### PR TITLE
infa: add python fuzz-introspector support

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Adds necessary logic for running fuzz-introspector's Python frontend.

Signed-off-by: David Korczynski <david@adalogics.com>

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
Ref: https://github.com/ossf/fuzz-introspector/issues/280

**Review by Navidem — APPROVED** [review]
LGTM

**Comment by jonathanmetzman** [comment]
@Navidem OK I don't need trial builds now, please test this before landing.

File: infra/base-images/base-builder/compile
Comment (Navidem) [inline infra/base-images/base-builder/compile:46]: nit: Error message is missing Introspector as a valid sanitizer.

File: infra/base-images/base-builder/compile_python_fuzzer
Comment (Navidem) [inline infra/base-images/base-builder/compile_python_fuzzer:44]: Does not go beyond this once `pycg` is install, here is the error:
```
INFO:__main__:Starting analysis
INFO:__main__:Running analysis with arguments: {fuzzer: /src/fuzz_glom.py, package: /src/glom/glom/ }
Traceback (most recent call last):
  File "/fuzz-introspector/frontends/python/main.py", line 354, in <module>
    sys.exit(main())
  File "/fuzz-introspector/frontends/python/main.py", line 342, in main
    exit_code = run_fuzz_pass(
  File "/fuzz-introspector/frontends/python/main.py", line 100, in run_fuzz_pass
    formatter = formats.Fuzz(cg)
AttributeError: module 'pycg.formats' has no attribute 'Fuzz'
```

Comment (Navidem) [reply infra/base-images/base-builder/compile_python_fuzzer:44]: Checking `pycg` [repo](https://github.com/vitsalis/PyCG), there is no such `pycg.formats.Fuzz`.  Maybe it is your own developed module?

**Review by Navidem — CHANGES_REQUESTED** [review]
Now blocked on `base-clang/Dockerfile`.

File: infra/base-images/base-builder/compile_python_fuzzer
Comment (DavidKorczynski) [reply infra/base-images/base-builder/compile_python_fuzzer:44]: Did you rebiuld `base-clang`? This is needed becaues you need the new version of fuzz-introspector that links to the custom PyCG -- see https://github.com/ossf/fuzz-introspector/tree/main/frontends/python

Comment (Navidem) [reply infra/base-images/base-builder/compile_python_fuzzer:44]: Ah! that's the missing step, I just rebuilt `base-builder` and `base-builder-python`. Thanks for prompt response!

**Review by Navidem — CHANGES_REQUESTED** [review]
Once `base-clang/Dockerfile` fixed, it successfully finishes.

File: infra/base-images/base-builder/compile_python_fuzzer
Comment (Navidem) [reply infra/base-images/base-builder/compile_python_fuzzer:36]: nm.

File: infra/base-images/base-clang/Dockerfile
Comment (DavidKorczynski) [reply infra/base-images/base-clang/Dockerfile]: Thanks a lot!

File: infra/base-images/base-clang/Dockerfile
Comment (DavidKorczynski) [reply infra/base-images/base-clang/Dockerfile]: Thanks a lot!

**Comment by Navidem** [comment]
/gcbrun trial_build.py glom idna pyyaml --sanitizer introspector --fuzzing-engine libfuzzer

File: infra/base-images/base-builder/compile
Comment (DavidKorczynski) [reply infra/base-images/base-builder/compile]: Done!

**Comment by DavidKorczynski** [comment]
Is this ready for merge?

**Comment by Navidem** [comment]
/gcbrun trial_build.py glom idna pyyaml --sanitizer introspector --fuzzing-engine libfuzzer --force

**Comment by Navidem** [comment]
> Is this ready for merge?

Looks good to me, I would wait for trial build to succeed.

**Comment by DavidKorczynski** [comment]
The change bumps fuzz-introspector, and thus does impact base-clang now that fuzz-introspector is in the main clang. Does it make sense to test more than the three Python projects just to be extra cautious?

**Comment by Navidem** [comment]
> The change bumps fuzz-introspector, and thus does impact base-clang now that fuzz-introspector is in the main clang. Does it make sense to test more than the three Python projects just to be extra cautious?

Good point! Let's wait for the python ones to finish then I run another trial.

**Comment by Navidem** [comment]
/gcbrun trial_build.py htslib bzip2 aspell boringssl --sanitizer introspector --fuzzing-engine libfuzzer --force

**Comment by Navidem** [comment]
All trials finished successfully. Good to merge.

**Comment by DavidKorczynski** [comment]
> All trials finished successfully. Good to merge.

Awesome. I can't merge this myself but am happy to get this in.

**Pull request merged**
